lib/all_changed_attributes.rb

```diff
@@ -0,0 +1,47 @@
+require 'all_changed_attributes/version'
+
+module AllChangedAttributes
+  def write_attribute(attr_name, value)
+    store_all_changed_attributes(attr_name, value) if attributes[attr_name] != value
+    super
+  end
+
+  def all_changed_attributes(new = {})
+    new.merge(@all_changed_attributes ||= {})
+  end
+
+  def store_all_changed_attributes(attr_name, value)
+    @all_changed_attributes = all_changed_attributes({attr_name.to_s => attributes[attr_name]})
```

Comment:
If an attribute is changed multiple times in a transaction, then it looks like this won't store the original value, it will store the second last value.